pkgs/tools/networking/labctl/default.nix

```diff
@@ -0,0 +1,41 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, installShellFiles
+}:
+
+buildGoModule rec {
+  pname = "labctl";
+  version = "0.0.15";
+
+  src = fetchFromGitHub {
+    owner = "labctl";
+    repo = "labctl";
+    rev = "v${version}";
+    hash = "sha256-txleZMgj/06PmP8Bv1J6n/2ywViNFqlgdCNzaTrY58c=";
+  };
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  vendorHash = "sha256-JXFw/u8QaZKt+OjUWW0rPi9QDSkiqYyqcNCxyzvcDM4=";
+
+  ldflags = [
+    "-X=github.com/labctl/labctl/app.version=${version}"
+    "-X=github.com/labctl/labctl/app.commit=${src.rev}"
+    "-X=github.com/labctl/labctl/app.date=1970-01-01T00:00:00Z"
+  ];
+
+  postInstall = ''
+    local INSTALL="$out/bin/labctl"
+    installShellCompletion --cmd labctl \
+      --bash <(echo "complete -C $INSTALL labctl") \
+      --zsh <(echo "complete -o nospace -C $INSTALL labctl")
```

Comment:
that doesn't look right, is this how zsh completions are supposed to be added?

Reply:
![image](https://github.com/NixOS/nixpkgs/assets/80165193/5632f819-7524-4133-ba68-f037db3dee61)
![image](https://github.com/NixOS/nixpkgs/assets/80165193/35e27af3-557c-4403-9e81-43654b9586c6)

It seems to work fine. 
ZSH seems to have a function for complete:
```
which complete
complete () {
	emulate -L zsh
	local args void cmd print remove
	args=("$@") 
	zparseopts -D -a void o: A: G: W: C: F: P: S: X: a b c d e f g j k u v p=print r=remove
	if [[ -n $print ]]
	then
		printf 'complete %2$s %1$s\n' "${(@kv)_comps[(R)_bash*]#* }"
	elif [[ -n $remove ]]
	then
		for cmd
		do
			unset "_comps[$cmd]"
		done
	else
		compdef _bash_complete\ ${(j. .)${(q)args[1,-1-$#]}} "$@"
	fi
}
```

I used the same thing the [waypoint](https://github.com/NixOS/nixpkgs/blob/e10802309bf9ae351eb27002c85cfdeb1be3b262/pkgs/applications/networking/cluster/waypoint/default.nix#L27-L52) derivation is implementing. 

Okay just installed waypoint and opend a new shell 
![image](https://github.com/NixOS/nixpkgs/assets/80165193/f016f244-87f5-4128-893b-b887e1cef831)
zsh seems broken their because it dosn't exececute the command, executing the content of the file in ${waypoint}/share/zsh/site-functions/_labctl works fine and makes completion work.
Bash works fine 
![image](https://github.com/NixOS/nixpkgs/assets/80165193/4cd350bb-0bb2-49a7-a370-6e8b8300b3a2)

Reply:
I'll take your word for it. I think it might have something to do with the fact that I'm ran an unwrapped zsh

Reply:
ahhh no, I said it currently seems broken

>  zsh seems broken their because it dosn't exececute the command, executing the content of the file in ${waypoint}/share/zsh/site-functions/_labctl works fine and makes completion work.

Reply:
sorry :(

Reply:
oh oops I misread it

Reply:
I guess it's fine for now since it doesn't break anything, but please open a follow-up PR when you find out what went wrong

btw you might want to look into `compdef`, and the file might need to have something like `#compdef labctl` at the top